src/openforms/submissions/tests/test_submission_status.py

```diff
@@ -21,13 +21,18 @@
 from ..constants import SUBMISSIONS_SESSION_KEY, ProcessingResults, ProcessingStatuses
 from ..tasks import cleanup_on_completion_results
 from ..tokens import submission_status_token_generator
-from .factories import SubmissionFactory, SubmissionReportFactory
+from .factories import (
+    PostCompletionTasksIDsFactory,
+    SubmissionFactory,
+    SubmissionReportFactory,
+)
 
 
 class SubmissionStatusPermissionTests(APITestCase):
     def test_valid_token(self):
         # Use empty task ID to not need a real broker
-        submission = SubmissionFactory.create(completed=True, on_completion_task_ids=[])
+        submission = SubmissionFactory.create(completed=True)
+        PostCompletionTasksIDsFactory.create(submission=submission, tasks_ids=[])
```

Comment:
I'd consider putting the related model creation inside the `completed` trait too.